content/post/2018-04-02-sct-encoding.markdown

```diff
@@ -0,0 +1,439 @@
+---
+author: Jacob Hoffman-Andrews, Boulder developer
+date: 2018-04-02T00:00:00Z
+excerpt: How Signed Certificate Timestamps get embedded in certificates
+title: "Engineering deep dive: Encoding of SCTs in certificates"
+slug: sct-encoding
+---
+
+Let's Encrypt recently [launched SCT embedding in
+certificates](https://community.letsencrypt.org/t/signed-certificate-timestamps-embedded-in-certificates/57187).
+This feature allows browsers to check that a certificate was submitted to a
+Certificate Transparency log. As part of the launch, we did a thorough review
+that the encoding of Signed Certificate Timestamps (SCTs) in our certificates
+matches the relevant specifications. In this post, I'll dive into the details.
+You'll learn more about X.509, ASN.1, DER, and TLS encoding, with references to
+the relevant RFCs.
+
+Certificate Transparency offers three ways to deliver SCTs to a browser: In a
+TLS extension, in stapled OCSP, or embedded in a certificate. We chose to
+implement the embedding method because it would just work for Let's Encrypt
+subscribers without additional work. In the SCT embedding method, we submit
+a "precertificate" with a [poison extension](#poison) to a set of
+CT logs, and get back SCTs. We then issue a real certificate based on the
+precertificate, with two changes: The poison extension is removed, and a list
+of SCTs is added in another extension.
+
+Given a certificate, let's first look for the extension. According to CT ([RFC 6962
+section 3.3](https://tools.ietf.org/html/rfc6962#section-3.3)),
+the extension OID for a list of SCTs is 1.3.6.1.4.1.11129.2.4.2. An [OID (object
+ID)](http://www.hl7.org/Oid/information.cfm) is a series of integers, hierarchically
+assigned and globally unique. They are used extensively in X.509, for instance
+to uniquely identify extensions.
+
+We can [download an example certificate](https://acme-v01.api.letsencrypt.org/acme/cert/031f2484307c9bc511b3123cb236a480d451),
+and view it using OpenSSL (if your OpenSSL is old, it may not display the
```

Comment:
"If your OpenSSL is old" - Can we give more specific advice here? Older than 1.0.1?